src/Datadog.Trace.ClrProfiler.Native/il_rewriter.cpp

```diff
@@ -176,6 +176,13 @@ EHClause* ILRewriter::GetEHPointer()
 
 void ILRewriter::SetEHClause(EHClause* ehPointer, unsigned ehLength)
 {
+    if (m_pEH != nullptr)
+    {
+        // Delete previous array
+        m_nEH = 0;
```

Comment:
you could use a `std::unique_ptr` for `m_nEh` and use the `reset` method to i) replace the old array by the new array ii) release the memory of the old array.

Reply:
I thought about it, but decide to don't change this `il_rewiter.cpp` file a lot, because the `m_nEH` member is not the only pointer. Maybe I can leave that for a bigger refactor of that class.